packages/widgets/src/Pagination/PageSummary.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import propTypes from '@dhis2/prop-types'
+import i18n from '@dhis2/d2-i18n'
+import { spacers } from '@dhis2/ui-constants'
+
+import { sectionStyles } from './Pagination.styles.js'
+
+const getItemRange = (page, pageSize, total) => {
+    const firstItem = total === 0 ? 0 : (page - 1) * pageSize + 1
+    const lastItem = Math.min(firstItem + pageSize - 1, total)
+
+    return { firstItem, lastItem }
+}
+
+const PageSummary = ({ dataTest, page, pageCount, pageSize, total }) => {
+    const { firstItem, lastItem } = getItemRange(page, pageSize, total)
+
+    return (
+        <div className="section" data-test={`${dataTest}-summary`}>
```

Comment:
If I remember correctly, the preferred approach is to supply the data-test values in the root component. I can see that we have mixed approaches in our codebase, so I'm not sure I remember the correct variant. @ismay do you remember? I think you were the one advancing a standard here. It's currently a convention so I'd prefer to not address it until we have a linter plugin or so, but I don't know how strongly you feel about this

Reply:
> to supply the data-test values in the root component.

I thought this was the preferred approach for internal components. The root gets the base `dataTest` value and the internal components get values derived from that.

Reply:
Yeah what Hendrik is saying. https://github.com/dhis2/ui/blob/master/packages/core/src/MultiSelect/MultiSelect.js#L97

Though I see it's not the case everywhere. But the above was the strategy when I originally added them.